File: src/levels/L08-facade/base_precalentar.rb
class FrontDesk
  # region: FrontDesk#make_latte
  def make_latte
    ground = Grinder.new.grind(:arabica)
    machine = EspressoMachine.new
    machine.preheat # ← cambio 1
    shot = machine.extract(ground)
    milk = MilkFrother.new.froth(:entera)
    Counter.new.hand_over([shot, milk])
  end
  # endregion
end

class DeliveryApp
  # region: DeliveryApp#make_latte
  def make_latte
    ground = Grinder.new.grind(:arabica)
    machine = EspressoMachine.new
    machine.preheat # ← cambio 2: la misma edición, en otro lugar
    shot = machine.extract(ground)
    Counter.new.hand_over([shot])
  end
  # endregion
end

File: src/levels/L08-facade/ts/base_precalentar.ts
class FrontDesk {
  // region: FrontDesk#make_latte
  makeLatte() {
    const ground = new Grinder().grind('arabica')
    const machine = new EspressoMachine()
    machine.preheat() // ← cambio 1
    const shot = machine.extract(ground)
    const milk = new MilkFrother().froth('entera')
    new Counter().handOver([shot, milk])
  }
  // endregion
}

class DeliveryApp {
  // region: DeliveryApp#make_latte
  makeLatte() {
    const ground = new Grinder().grind('arabica')
    const machine = new EspressoMachine()
    machine.preheat() // ← cambio 2: la misma edición, en otro lugar
    const shot = machine.extract(ground)
    new Counter().handOver([shot])
  }
  // endregion
}
